app/grants/tasks.py

```diff
@@ -118,78 +120,202 @@ def process_grant_contribution(self, grant_id, grant_slug, profile_id, package,
+    for (index, grant_id) in enumerate(grant_ids_list):
+        try:
+            grant = Grant.objects.get(pk=grant_id)
+        except Grant.DoesNotExist:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Does Not Exist'),
+                'grant':grant_id,
+                'text': _('This grant does not exist'),
+                'subtext': _('No grant with this ID was found'),
+                'success': False
+            })
+            continue
+
+        if not grant.active:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Ended'),
+                'grant':grant_id,
+                'text': _('This Grant has ended.'),
+                'subtext': _('Contributions can no longer be made this grant'),
+                'success': False
+            })
+            continue
+
+        if is_grant_team_member(grant, profile):
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant funding blocked'),
+                'grant':grant_id,
+                'text': _('This Grant cannot be funded'),
+                'subtext': _('Grant team members cannot contribute to their own grant.'),
+                'success': False
+            })
+            continue
+
+        if grant.link_to_new_grant:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Migrated'),
+                'grant':grant_id,
+                'text': _('This Grant has ended'),
+                'subtext': _('Contributions can no longer be made to this grant. <br> Visit the new grant to contribute.'),
+                'success': False
+            })
+            continue
+
+        active_subscription = Subscription.objects.select_related('grant').filter(
+            grant=grant_id, active=True, error=False, contributor_profile=profile, is_postive_vote=True
+        )
+
+        if active_subscription:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Subscription Exists'),
+                'grant':grant_id,
+                'text': _('You already have an active subscription for this grant.'),
+                'success': False
+            })
+            continue
+
+        if not grant.configured_to_receieve_funding:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Not Configured'),
+                'grant':grant_id,
+                'text': _('This Grant is not configured to accept funding at this time.'),
+                'subtext': _('Grant is not properly configured for funding.  Please set grant.contract_address on this grant, or contact founders@gitcoin.co if you believe this message is in error!'),
+                'success': False
+            })
+            continue
+
+        try:
+            package = {
```

Comment:
it seems pointless to create this object here! wouls it make sense to create this outside while we do this 
https://github.com/gitcoinco/web/pull/8018/files#r537294299